<!DOCTYPE html>
<html lang="en-us" dir="ltr">
<head>
  <meta property="og:title" content="Differential abundance analysis" />
<meta property="og:description" content="I borrow the data from this article ( https://www.pnas.org/doi/10.1073/pnas.2008762117#data-availability) to analyze Alzheimerâ€™s disease (AD) data, aiming to show the standard analysis during my work. The previous blog mainly focused on the quality control, dimensionality reduction, clustering, and cell type annotation of single-nuclei RNA seq data. This blog further analyzes the data for differential abundance analysis.
Libraries # library(Seurat) library(dplyr) library(ggplot2) library(edgeR) Abundance visualization # seu &lt;- readRDS(&#34;../../../../../transcriptome_protein_AD_plat/sc_RNA/Seu_PCA_UMAP_annotated.RDS&#34;) Seurat::DimPlot(seu, reduction=&#34;umap&#34;,label = TRUE)&#43;NoLegend() Seurat::DimPlot(seu, reduction=&#34;umap&#34;, group." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/" /><meta property="og:image" content="https://xiwen514.github.io/47"/><meta property="article:section" content="post" />
<meta property="article:published_time" content="2023-10-30T00:00:00+00:00" />
<meta property="article:modified_time" content="2023-10-30T00:00:00+00:00" /><meta property="og:site_name" content="ðŸ““ Sivan&#39;s blog" />
<title>Differential abundance analysis | ðŸ““ Sivan&#39;s blog</title>
<link rel="manifest" href="/manifest.json">
<link rel="icon" href="/favicon.png" type="image/x-icon">
<link rel="stylesheet" href="/book.min.45ec1f6261e17c6488824ac77b6b8a0c11eef38eaed587cfecad4b284f57ee79.css" integrity="sha256-RewfYmHhfGSIgkrHe2uKDBHu846u1YfP7K1LKE9X7nk=" crossorigin="anonymous">

<!--
Made with Book Theme
https://github.com/alex-shpak/hugo-book
-->
  
  



<meta charset="UTF-8" />
<meta
  name="description"
  content="I borrow the data from this article ( https://www.pnas.org/doi/10.1073/pnas.2008762117#data-availability) to analyze Alzheimerâ€™s disease (AD) data, aiming to show the standard analysis during my work. The previous blog mainly focused on the quality control, dimensionality reduction, clustering, and cell type annotation of single-nuclei RNA seq data. This blog further analyzes the data for differential abundance analysis.
Libraries # library(Seurat) library(dplyr) library(ggplot2) library(edgeR) Abundance visualization # seu &lt;- readRDS(&#34;../../../../../transcriptome_protein_AD_plat/sc_RNA/Seu_PCA_UMAP_annotated.RDS&#34;) Seurat::DimPlot(seu, reduction=&#34;umap&#34;,label = TRUE)&#43;NoLegend() Seurat::DimPlot(seu, reduction=&#34;umap&#34;, group."
/>
<title>
  Differential abundance analysis
</title>
<meta name="viewport" content="width=device-width, initial-scale=1" />






<link rel="shortcut icon" type="image/png"  href="https://xiwen514.github.io//favicon.png" />

























  
  


<link href="https://xiwen514.github.io/styles.e08ceb33360cec132feb69cfb982e2a4.min.css" rel="stylesheet" />


<link href="https://xiwen514.github.io/quartz/styles/_light_syntax.86a48a52faebeaaf42158b72922b1c90.min.css" rel="stylesheet" id="theme-link">

  

<link rel="preload" href="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css" as="style"
      onload="this.onload=null;this.rel='stylesheet'"
      integrity="sha384-R4558gYOUz8mP9YWpZJjofhk+zx0AS11p36HnD2ZKj/6JR5z27gSSULCNHIRReVs" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js"
        integrity="sha384-z1fJDqw8ZApjGO3/unPWUPsIymfsJmyrDVWC8Tv/a1HeOtGmkwNd/7xUS0Xcnvsx"
        crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/contrib/auto-render.min.js"
        integrity="sha384-+XBljXPPiv+OzfbB3cVmLHf4hdUFHlWNZN5spNQ7rmHTXpd7WvJum6fIACpNNfIR"
        crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.2/dist/contrib/copy-tex.min.js"
        integrity="sha384-ww/583aHhxWkz5DEVn6OKtNiIaLi2iBRNZXfJRiY1Ai7tnJ9UXpEsyvOITVpTl4A"
        crossorigin="anonymous"></script>

  






<script src="https://xiwen514.github.io/quartz/js/darkmode.b187bc80fd41724185e768ddede977f5.min.js"></script>



<script src="https://xiwen514.github.io/quartz/js/util.c69e233dc1cc331a30b0f670e657b425.min.js"></script>



<script async src="https://unpkg.com/@floating-ui/core@0.7.3"></script>
<script async src="https://unpkg.com/@floating-ui/dom@0.5.4"></script>

<script src="https://xiwen514.github.io/quartz/js/popover.f03552ccb84d99ca615d1cfb9abde59e.min.js"></script>




<script defer src="https://xiwen514.github.io/quartz/js/code-title.ce4a43f09239a9efb48fee342e8ef2df.min.js"></script>




<script defer src="https://xiwen514.github.io/quartz/js/clipboard.2913da76d3cb21c5deaa4bae7da38c9f.min.js"></script>




<script defer src="https://xiwen514.github.io/quartz/js/callouts.7723cac461d613d118ee8bb8216b9838.min.js"></script>



 
<script>
  const SEARCH_ENABLED =  null 
  const LATEX_ENABLED =  true 
  const PRODUCTION =  true 
  const BASE_URL = "https://xiwen514.github.io/"
  const fetchData = Promise.all([
        fetch("https:\/\/xiwen514.github.io\/indices\/linkIndex.f6425e3c9c5a78cb1a57d6271bd0c88d.min.json")
          .then(data => data.json())
          .then(data => ({
            index: data.index,
            links: data.links,
          })),
        fetch("https:\/\/xiwen514.github.io\/indices\/contentIndex.56753ef950db2332e50166167e5b613e.min.json")
          .then(data => data.json()),
      ])
      .then(([{index, links}, content]) => ({
        index,
        links,
        content,
      }))

    const render = () => {
    
    const siteBaseURL = new URL(BASE_URL);
    const pathBase = siteBaseURL.pathname;
    const pathWindow = window.location.pathname;
    const isHome = pathBase == pathWindow;

    addCopyButtons();
    

    

    addCollapsibleCallouts();
    

    
    initPopover(
      "https://xiwen514.github.io",
       false 
    )
    

    

    
      var els = document.getElementsByClassName("mermaid");
      if (els.length > 0) {
        import('https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs').then(
          (obj) => {
            
            
            obj.default.init();
          }
        )
      }
    
  }

  const init = (doc = document) => {
    
    addCopyButtons();
    

    addTitleToCodeBlocks();
    renderMathInElement(doc.body, {
      delimiters: [
        {left: '$$', right: '$$', display: true},
        {left: '$', right: '$', display: false},
      ],
      macros: {
        'â€™': "'"
      },
      throwOnError : false
    });
    
  };
</script>

<script>
  window.Million = {
    navigate: (url) => (window.location.href = url),
    prefetch: () => {},
  }

  window.addEventListener("DOMContentLoaded", () => {
    init()
    render()
  })
</script>


</head>
<div id="search-container">
  <div id="search-space">
    <input autocomplete="off" id="search-bar" name="search" type="text" aria-label="Search"
      placeholder="Search for something..." dir="">
    <div id="results-container">
    </div>
  </div>
</div>

<script src="https://cdn.jsdelivr.net/npm/flexsearch@0.7.21/dist/flexsearch.bundle.js"
  integrity="sha256-i3A0NZGkhsKjVMzFxv3ksk0DZh3aXqu0l49Bbh0MdjE=" crossorigin="anonymous" defer></script>

<script defer src="https://xiwen514.github.io/quartz/js/full-text-search.e6e2e0c213187ca0c703d6e2c7a77fcd.min.js"></script>



<header class="book-header">
    
    <label for="menu-control">
      <img src="/svg/menu.svg" class="book-icon" alt="Menu" />
    </label>
    <h1 id="page-title"><a href="https://xiwen514.github.io/">ðŸ““ Sivan&#39;s blog</a></h1>
    <div class="spacer"></div>
    <div id="search-icon" class="quartz-search">
      <p>Search</p>
      <svg tabindex="0" aria-labelledby="title desc" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19.9 19.7"><title id="title">Search Icon</title><desc id="desc">Icon to open search</desc><g class="search-path" fill="none"><path stroke-linecap="square" d="M18.5 18.3l-5.4-5.4"/><circle cx="8" cy="8" r="7"/></g></svg>
    </div>
    <div class='darkmode'>
    <input class='toggle' id='darkmode-toggle' type='checkbox' tabindex="-1">
    <label id="toggle-label-light" for='darkmode-toggle' tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="dayIcon" x="0px" y="0px" viewBox="0 0 35 35" style="enable-background:new 0 0 35 35;" xml:space="preserve">
            <title>Light Mode</title>
            <path d="M6,17.5C6,16.672,5.328,16,4.5,16h-3C0.672,16,0,16.672,0,17.5    S0.672,19,1.5,19h3C5.328,19,6,18.328,6,17.5z M7.5,26c-0.414,0-0.789,0.168-1.061,0.439l-2,2C4.168,28.711,4,29.086,4,29.5    C4,30.328,4.671,31,5.5,31c0.414,0,0.789-0.168,1.06-0.44l2-2C8.832,28.289,9,27.914,9,27.5C9,26.672,8.329,26,7.5,26z M17.5,6    C18.329,6,19,5.328,19,4.5v-3C19,0.672,18.329,0,17.5,0S16,0.672,16,1.5v3C16,5.328,16.671,6,17.5,6z M27.5,9    c0.414,0,0.789-0.168,1.06-0.439l2-2C30.832,6.289,31,5.914,31,5.5C31,4.672,30.329,4,29.5,4c-0.414,0-0.789,0.168-1.061,0.44    l-2,2C26.168,6.711,26,7.086,26,7.5C26,8.328,26.671,9,27.5,9z M6.439,8.561C6.711,8.832,7.086,9,7.5,9C8.328,9,9,8.328,9,7.5    c0-0.414-0.168-0.789-0.439-1.061l-2-2C6.289,4.168,5.914,4,5.5,4C4.672,4,4,4.672,4,5.5c0,0.414,0.168,0.789,0.439,1.06    L6.439,8.561z M33.5,16h-3c-0.828,0-1.5,0.672-1.5,1.5s0.672,1.5,1.5,1.5h3c0.828,0,1.5-0.672,1.5-1.5S34.328,16,33.5,16z     M28.561,26.439C28.289,26.168,27.914,26,27.5,26c-0.828,0-1.5,0.672-1.5,1.5c0,0.414,0.168,0.789,0.439,1.06l2,2    C28.711,30.832,29.086,31,29.5,31c0.828,0,1.5-0.672,1.5-1.5c0-0.414-0.168-0.789-0.439-1.061L28.561,26.439z M17.5,29    c-0.829,0-1.5,0.672-1.5,1.5v3c0,0.828,0.671,1.5,1.5,1.5s1.5-0.672,1.5-1.5v-3C19,29.672,18.329,29,17.5,29z M17.5,7    C11.71,7,7,11.71,7,17.5S11.71,28,17.5,28S28,23.29,28,17.5S23.29,7,17.5,7z M17.5,25c-4.136,0-7.5-3.364-7.5-7.5    c0-4.136,3.364-7.5,7.5-7.5c4.136,0,7.5,3.364,7.5,7.5C25,21.636,21.636,25,17.5,25z" />
        </svg>
    </label>
    <label id="toggle-label-dark" for='darkmode-toggle' tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="nightIcon" x="0px" y="0px" viewBox="0 0 100 100" style="enable-background='new 0 0 100 100'" xml:space="preserve">
            <title>Dark Mode</title>
            <path d="M96.76,66.458c-0.853-0.852-2.15-1.064-3.23-0.534c-6.063,2.991-12.858,4.571-19.655,4.571  C62.022,70.495,50.88,65.88,42.5,57.5C29.043,44.043,25.658,23.536,34.076,6.47c0.532-1.08,0.318-2.379-0.534-3.23  c-0.851-0.852-2.15-1.064-3.23-0.534c-4.918,2.427-9.375,5.619-13.246,9.491c-9.447,9.447-14.65,22.008-14.65,35.369  c0,13.36,5.203,25.921,14.65,35.368s22.008,14.65,35.368,14.65c13.361,0,25.921-5.203,35.369-14.65  c3.872-3.871,7.064-8.328,9.491-13.246C97.826,68.608,97.611,67.309,96.76,66.458z" />
        </svg>
    </label>
</div>
</header>


<body dir="ltr">
  <input type="checkbox" class="hidden toggle" id="menu-control" />
  <main class="flex">
    <aside class="book-menu">
      <div class="book-menu-content">
        
  <nav>
<div class="menu-search">
  <div id="search-icon" class="quartz-search">
    <p>Search</p>
    <svg tabindex="0" aria-labelledby="title desc" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19.9 19.7">
      <title id="title">Search Icon</title>
      <desc id="desc">Icon to open search</desc>
      <g class="search-path" fill="none">
        <path stroke-linecap="square" d="M18.5 18.3l-5.4-5.4" />
        <circle cx="8" cy="8" r="7" />
      </g>
    </svg>
  </div>
</div>













  



  
  <ul>
    
      
        <li>
          
  
  

  
    <span class="book-menu-title">Posts</span>
  

          
  <ul>
    
      
        <li>
          
  
  

  
    <a href="/post/2023-10-30-diff-abun-analysis/" class="active">Differential abundance analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-10-25-single-nucleus-analysis/" class="">Single nucleus analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-10-23-multi-omics-intergration-of-the-platelet-in-alzheimer-s-disease/" class="">Intergration of the platelet transcriptome and proteome in Alzheimerâ€™s disease and aging data analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-10-23-the-platelet-transcriptome-in-alzheimer-s-disease/" class="">The platelet transcriptome in Alzheimerâ€™s disease and aging data analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-10-23-the-platelet-proteome-in-alzheimer-s-disease/" class="">The platelet proteome in Alzheimerâ€™s disease and aging data analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-09-29-rppa-data-analysis/" class="">RPPA data analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-09-14-functional-analysis/" class="">Functional analysis</a>
  

        </li>
      
    
      
        <li>
          
  
  

  
    <a href="/post/2023-09-14-introduction-to-pca/" class="">Introduction to PCA</a>
  

        </li>
      
    
  </ul>

        </li>
      
    
  </ul>











  
<ul>
  
  <li>
    <a href="https://github.com/xiwen514/xiwen514.github.io"  target="_blank" rel="noopener">
        Github
      </a>
  </li>
  
</ul>






</nav>




  <script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script>


 
      </div>
    </aside>
    <div class="book-page container">
      

      
      
  <article class="markdown">
    <h1 class="title">Differential abundance analysis</h1>
    






  
  

  
  

  
  

  
  

  
  

  
  

  
  















  
  
  
  
  
  
  
  
  
  






<p>I borrow the data from this article (
<a href="https://www.pnas.org/doi/10.1073/pnas.2008762117#data-availability" rel="noopener">https://www.pnas.org/doi/10.1073/pnas.2008762117#data-availability</a>) to analyze Alzheimerâ€™s disease (AD) data, aiming to show the standard analysis during my work. The previous blog mainly focused on the quality control, dimensionality reduction, clustering, and cell type annotation of single-nuclei RNA seq data. This blog further analyzes the data for differential abundance analysis.</p>
<h3 id="libraries">
  Libraries
  <a class="anchor" href="#libraries">#</a>
</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">library</span>(Seurat)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">library</span>(dplyr)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">library</span>(ggplot2)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">library</span>(edgeR)
</span></span></code></pre></div><h3 id="abundance-visualization">
  Abundance visualization
  <a class="anchor" href="#abundance-visualization">#</a>
</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>seu <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">readRDS</span>(<span style="color:#f1fa8c">&#34;../../../../../transcriptome_protein_AD_plat/sc_RNA/Seu_PCA_UMAP_annotated.RDS&#34;</span>)
</span></span><span style="display:flex;"><span>Seurat<span style="color:#ff79c6">::</span><span style="color:#50fa7b">DimPlot</span>(seu, reduction<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;umap&#34;</span>,label <span style="color:#ff79c6">=</span> <span style="color:#ff79c6">TRUE</span>)<span style="color:#ff79c6">+</span><span style="color:#50fa7b">NoLegend</span>()
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-2-1.png" width="672" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>Seurat<span style="color:#ff79c6">::</span><span style="color:#50fa7b">DimPlot</span>(seu, reduction<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;umap&#34;</span>, group.by<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;group_id&#34;</span>)
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-2-2.png" width="672" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#6272a4"># Create single cell experiment object</span>
</span></span><span style="display:flex;"><span>sce <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">as.SingleCellExperiment</span>(seu, assays<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;RNA&#34;</span>)
</span></span></code></pre></div><pre tabindex="0"><code>## Warning: The following arguments are not used: assays
</code></pre><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>n_cells <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">table</span>(sce<span style="color:#ff79c6">$</span>cell_type, sce<span style="color:#ff79c6">$</span>orig.ident) 
</span></span><span style="display:flex;"><span>freqs <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">prop.table</span>(n_cells, margin <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">2</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>coldata <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">data.frame</span>(sce<span style="color:#ff79c6">@</span>colData)
</span></span><span style="display:flex;"><span>coldata <span style="color:#ff79c6">&lt;-</span> coldata[<span style="color:#ff79c6">!</span><span style="color:#50fa7b">duplicated</span>(coldata<span style="color:#ff79c6">$</span>orig.ident),]
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">colnames</span>(coldata)[1] <span style="color:#ff79c6">&lt;-</span> <span style="color:#f1fa8c">&#34;Sample&#34;</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>df <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">as.data.frame</span>(<span style="color:#50fa7b">t</span>(freqs))
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">colnames</span>(df)[1<span style="color:#ff79c6">:</span><span style="color:#bd93f9">3</span>] <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">c</span>(<span style="color:#f1fa8c">&#34;Sample&#34;</span>, <span style="color:#f1fa8c">&#34;cluster_id&#34;</span>, <span style="color:#f1fa8c">&#34;frequency&#34;</span>)
</span></span><span style="display:flex;"><span>df <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">left_join</span>(df, coldata[,<span style="color:#50fa7b">c</span>(<span style="color:#f1fa8c">&#34;Sample&#34;</span>, <span style="color:#f1fa8c">&#34;group_id&#34;</span>)])
</span></span></code></pre></div><pre tabindex="0"><code>## Joining with `by = join_by(Sample)`
</code></pre><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">colnames</span>(df)[4] <span style="color:#ff79c6">&lt;-</span> <span style="color:#f1fa8c">&#34;diagnosis_id&#34;</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>df<span style="color:#ff79c6">$</span>diagnosis_id <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">factor</span>(df<span style="color:#ff79c6">$</span>diagnosis_id, levels <span style="color:#ff79c6">=</span> <span style="color:#50fa7b">c</span>(<span style="color:#f1fa8c">&#34;AD&#34;</span>,<span style="color:#f1fa8c">&#34;NC&#34;</span>))
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">ggplot</span>(df, <span style="color:#50fa7b">aes</span>(x <span style="color:#ff79c6">=</span> Sample, y <span style="color:#ff79c6">=</span> frequency, fill <span style="color:#ff79c6">=</span> cluster_id)) <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">geom_bar</span>(stat <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;identity&#34;</span>, position <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;fill&#34;</span>) <span style="color:#ff79c6">+</span> 
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">facet_wrap</span>(<span style="color:#ff79c6">~</span> diagnosis_id, scales <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;free_x&#34;</span>, ncol <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">4</span>) <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">theme_classic</span>() <span style="color:#ff79c6">+</span> 
</span></span><span style="display:flex;"><span>  <span style="color:#50fa7b">theme</span>(axis.text.x <span style="color:#ff79c6">=</span> <span style="color:#50fa7b">element_text</span>(angle <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">45</span>, hjust <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">1</span>))
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-3-1.png" width="672" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">ggplot</span>(df, <span style="color:#50fa7b">aes</span>(x <span style="color:#ff79c6">=</span> diagnosis_id, y <span style="color:#ff79c6">=</span> frequency, fill <span style="color:#ff79c6">=</span> cluster_id)) <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">geom_bar</span>(stat <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;identity&#34;</span>, position <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;fill&#34;</span>) <span style="color:#ff79c6">+</span> 
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">theme_classic</span>() <span style="color:#ff79c6">+</span> 
</span></span><span style="display:flex;"><span>  <span style="color:#50fa7b">theme</span>(axis.text.x <span style="color:#ff79c6">=</span> <span style="color:#50fa7b">element_text</span>(angle <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">45</span>, hjust <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">1</span>))
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-3-2.png" width="672" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">par</span>(mfrow <span style="color:#ff79c6">=</span> <span style="color:#50fa7b">c</span>(<span style="color:#bd93f9">4</span>, <span style="color:#bd93f9">2</span>))
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">ggplot</span>(df, <span style="color:#50fa7b">aes</span>(x <span style="color:#ff79c6">=</span> diagnosis_id, y <span style="color:#ff79c6">=</span> frequency, color <span style="color:#ff79c6">=</span> diagnosis_id)) <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">geom_boxplot</span>(outlier.colour <span style="color:#ff79c6">=</span> <span style="color:#ff79c6">NA</span>) <span style="color:#ff79c6">+</span>  <span style="color:#50fa7b">geom_jitter</span>() <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">facet_wrap</span>(<span style="color:#ff79c6">~</span> cluster_id, scales <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;free_y&#34;</span>, ncol <span style="color:#ff79c6">=</span> <span style="color:#bd93f9">3</span>) <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">theme_classic</span>() <span style="color:#ff79c6">+</span>
</span></span><span style="display:flex;"><span>    <span style="color:#50fa7b">theme</span>(legend.position <span style="color:#ff79c6">=</span> <span style="color:#f1fa8c">&#34;none&#34;</span>)
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-3-3.png" width="672" />
<h3 id="use-edger-to-perform-differential-abundance-analysis">
  Use EdgeR to perform differential abundance analysis
  <a class="anchor" href="#use-edger-to-perform-differential-abundance-analysis">#</a>
</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>abundances <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">unclass</span>(n_cells) 
</span></span><span style="display:flex;"><span>abundances <span style="color:#ff79c6">&lt;-</span>  abundances[,<span style="color:#bd93f9">-4</span>] <span style="color:#6272a4"># delete NC11, assume NC11 as a outlier</span>
</span></span><span style="display:flex;"><span><span style="color:#6272a4">#Attaching some column metadata.</span>
</span></span><span style="display:flex;"><span>extra.info <span style="color:#ff79c6">&lt;-</span> sce<span style="color:#ff79c6">@</span>colData<span style="color:#50fa7b">[match</span>(<span style="color:#50fa7b">colnames</span>(abundances), sce<span style="color:#ff79c6">$</span>orig.ident),]
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#6272a4"># Creates a DGEList object from a table of counts</span>
</span></span><span style="display:flex;"><span>y.ab <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">DGEList</span>(abundances, samples<span style="color:#ff79c6">=</span>extra.info)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">summary</span>(y.ab<span style="color:#ff79c6">$</span>samples<span style="color:#ff79c6">$</span>lib.size)
</span></span></code></pre></div><pre tabindex="0"><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    3244    4237    4640    6438    5503   14568
</code></pre><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#6272a4"># Filter out low-abundance labels</span>
</span></span><span style="display:flex;"><span>keep <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">filterByExpr</span>(y.ab, group<span style="color:#ff79c6">=</span>y.ab<span style="color:#ff79c6">$</span>samples<span style="color:#ff79c6">$</span>seurat_clusters)
</span></span><span style="display:flex;"><span>y.ab <span style="color:#ff79c6">&lt;-</span> y.ab[keep,]
</span></span></code></pre></div><p>For a DA analysis of cluster abundances, filtering is generally not required as most clusters will not be of low-abundance (otherwise there would not have been enough evidence to define the cluster in the first place).</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>sample <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">as.factor</span>(y.ab<span style="color:#ff79c6">$</span>samples<span style="color:#ff79c6">$</span>orig.ident)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">plotMDS</span>(y.ab, pch<span style="color:#ff79c6">=</span><span style="color:#bd93f9">16</span>, col<span style="color:#ff79c6">=</span><span style="color:#50fa7b">c</span>(<span style="color:#bd93f9">2</span><span style="color:#ff79c6">:</span><span style="color:#bd93f9">8</span>)[sample], main<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;MDS&#34;</span>)
</span></span><span style="display:flex;"><span> <span style="color:#50fa7b">legend</span>(<span style="color:#f1fa8c">&#34;topright&#34;</span>, legend<span style="color:#ff79c6">=</span> <span style="color:#50fa7b">levels</span>(sample),pch<span style="color:#ff79c6">=</span><span style="color:#bd93f9">16</span>, col<span style="color:#ff79c6">=</span><span style="color:#bd93f9">2</span><span style="color:#ff79c6">:</span><span style="color:#bd93f9">8</span>, cex<span style="color:#ff79c6">=</span><span style="color:#bd93f9">0.8</span>)
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-5-1.png" width="672" />
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#6272a4"># Formulate the design matrix</span>
</span></span><span style="display:flex;"><span>design <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">model.matrix</span>(<span style="color:#ff79c6">~</span><span style="color:#50fa7b">factor</span>(group), y.ab<span style="color:#ff79c6">$</span>samples)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#6272a4"># Estimate the NB dispersion for each cluster</span>
</span></span><span style="display:flex;"><span>y.ab <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">estimateDisp</span>(y.ab, design, trend<span style="color:#ff79c6">=</span><span style="color:#f1fa8c">&#34;none&#34;</span>)
</span></span><span style="display:flex;"><span>y.ab<span style="color:#ff79c6">$</span>common.dispersion
</span></span></code></pre></div><pre tabindex="0"><code>## [1] 0.3259443
</code></pre><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">plotBCV</span>(y.ab)
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-5-2.png" width="672" />
<p>Biological coefficient of variation (BCV) for each label with respect to its average abundance. BCVs are defined as the square root of the NB dispersion. Common dispersion estimates are shown in red.</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#6272a4"># QL dispersion</span>
</span></span><span style="display:flex;"><span>fit.ab <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">glmQLFit</span>(y.ab, design, robust<span style="color:#ff79c6">=</span><span style="color:#ff79c6">TRUE</span>, abundance.trend<span style="color:#ff79c6">=</span><span style="color:#ff79c6">FALSE</span>)
</span></span><span style="display:flex;"><span><span style="color:#50fa7b">plotQLDisp</span>(fit.ab)
</span></span></code></pre></div><img src="https://xiwen514.github.io/post/2023-10-30-diff-abun-analysis/index_files/figure-html/unnamed-chunk-6-1.png" width="672" />
<p>QL dispersion estimates for each label with respect to its average abundance. Quarter-root values of the raw estimates are shown in black while the shrunken estimates are shown in red. Shrinkage is performed towards the common dispersion in blue.</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#6272a4"># Test for differences in abundance</span>
</span></span><span style="display:flex;"><span>res <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">glmQLFTest</span>(fit.ab, coef<span style="color:#ff79c6">=</span><span style="color:#50fa7b">ncol</span>(design))
</span></span><span style="display:flex;"><span>res_df <span style="color:#ff79c6">&lt;-</span> <span style="color:#50fa7b">topTags</span>(res, n <span style="color:#ff79c6">=</span> <span style="color:#ff79c6">Inf</span>)<span style="color:#ff79c6">$</span>table <span style="color:#ff79c6">%&gt;%</span> <span style="color:#50fa7b">round</span>(<span style="color:#bd93f9">2</span>)
</span></span><span style="display:flex;"><span>res_df
</span></span></code></pre></div><pre tabindex="0"><code>##                                logFC logCPM    F PValue  FDR
## Interneuron; GABAergic neuron   0.75  16.39 4.32   0.10 0.74
## Oligodendrocyte                -0.62  18.07 1.39   0.29 0.74
## Endothelial cell               -1.65  13.89 1.26   0.32 0.74
## Oligodendrocyte precursor cell  0.63  16.40 0.99   0.37 0.74
## Glutamatergic neuron            0.16  18.14 0.35   0.58 0.93
## Macrophage; Microglia cell      0.18  16.23 0.10   0.76 0.98
## Bergmann glial cell; Astrocyte -0.04  16.78 0.00   0.95 0.98
## GABAergic neuron                0.02  15.87 0.00   0.98 0.98
</code></pre><div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#282a36;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#50fa7b">summary</span>(<span style="color:#50fa7b">decideTests</span>(res))
</span></span></code></pre></div><pre tabindex="0"><code>##        factor(group)NC
## Down                 0
## NotSig               8
## Up                   0
</code></pre><p>None of clusters are significant between AD and NC.</p>


    

<hr/>


<div class="page-end" id="footer">
    <div class="backlinks-container">
        <h3>Backlinks</h3>
<ul class="backlinks">
    
    
    
    
    
    
    
    <li>
      No backlinks found
    </li>
    
</ul>

    </div>
    <div>
        <script
  src="https://cdn.jsdelivr.net/npm/d3@6.7.0/dist/d3.min.js"
  integrity="sha256-+7jaYCp29O1JusNWHaYtgUn6EhuP0VaFuswhNV06MyI="
  crossorigin="anonymous"
></script>
<h3>Interactive Graph</h3>
<div id="graph-container"></div>
<style>
  :root {
    --g-node: var(--secondary);
    --g-node-active: var(--primary);
    --g-node-inactive: var(--visited);
    --g-link: var(--outlinegray);
    --g-link-active: #5a7282;
  }
</style>

<script src="https://xiwen514.github.io/quartz/js/graph.cbd78cfa87df7d3e230d16fc24f06548.js"></script>

    </div>
</div>






<div id="contact_buttons">
    <footer>
        
        
        <p>Made with <a href="https://github.com/64bitpandas/amethyst">Amethyst</a>, Â© 2023 <no value></p>
        <ul>
            
            <li><a href="https://xiwen514.github.io/">Home</a></li>
            <li><a href="https://xiwen514.github.io/">Home</a></li><li><a href="https://www.linkedin.com/in/xiwen-zhao-a478581a3/">Linkdin</a></li></ul>
    </footer>
</div>


    
  </article>
 
      
      
      
      
  
  <div class="book-comments">

</div>
  
 

      <label for="menu-control" class="hidden book-menu-overlay"></label>
    </div>

    
    <aside class="book-toc">
      <div class="book-toc-content">
        
  
<nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#libraries">Libraries</a></li>
        <li><a href="#abundance-visualization">Abundance visualization</a></li>
        <li><a href="#use-edger-to-perform-differential-abundance-analysis">Use EdgeR to perform differential abundance analysis</a></li>
      </ul>
    </li>
  </ul>
</nav>


 
      </div>
    </aside>
    
  </main>

  
</body>
</html>












